animation_nodes/nodes/color/c_utils.pyx

```diff
@@ -0,0 +1,27 @@
+# cython: profile=True
+cimport cython
+
+from ... data_structures cimport (
+    ColorList,
+    VirtualColorList,
+    PolygonIndicesList
+)
+
+def loopsColorsList(long Amount, PolygonIndicesList polygons, VirtualColorList colors, 
+                    ColorList loopsColors):
+    cdef long i
+    for i in range(Amount):
+        loopsColors[i] = colors[polygons.indices[i]]
```

Comment:
Use the `get` method of VirtualList.

Reply:
Omar, I tired something like this `colors.get(polygons.indices[i])`. It gave an error `animation_nodes/nodes/color/c_utils.pyx:14:35: Cannot convert 'Color *' to Python object`.

Is there a way to find all the possible attributes of any object (e.g., possible attributes of Virtual List)? Maybe I'm wrong but the `dir()` function does not show any `get` attribute of Virtual Color List.

Reply:
`get` is a c method, you wouldn't see it with `dir` because it is not callable from Python. The error has nothing to do with the `get` method. It is probably caused by a wrong assignment, since `loopsColors[i]` is a python object. To assign to a `ColorList`, do `loopsColors.data[0] = colors.get(something)`.

Reply:
Here is the complete code,
```
from ... data_structures cimport (
    ColorList,
    VirtualColorList,
    PolygonIndicesList
)

def getLoopColorsFromVertexColors(PolygonIndicesList polygons, VirtualColorList colors):
    cdef long i
    cdef Amount = len(polygons.indices)
    cdef ColorList loopsColors = ColorList(length = Amount)

    for i in range(Amount):
        loopsColors.data[i] = colors.get(polygons.indices[i])
    return loopsColors
```
Stil, I'm getting the error,
![Screenshot from 2019-11-27 17-42-23](https://user-images.githubusercontent.com/30294746/69722536-6ac20980-113d-11ea-8f64-ada71a6ad123.png)

I have also tried this syntax but no success,
`loopsColors[i] = colors.get(polygons.indices[i])`

Reply:
Just dereference the right hand side. Dereferencing in Cython is done by `[0]`. For instance : `colors.get(polygons.indices[i])[0]`.

Reply:
Thanks, it worked. Should I also use this in second function like this `loopsColors.data[index] = colors.get(i)[0]`?

Reply:
Yes.